src/panels/lovelace/entity-rows/hui-input-number-entity-row.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  property,
+  customElement,
+  css,
+  CSSResult,
+} from "lit-element";
+import { PaperInputElement } from "@polymer/paper-input/paper-input";
+
+import "../components/hui-generic-entity-row";
+import "../../../components/ha-slider";
+import "../components/hui-warning";
+
+import { computeRTL } from "../../../common/util/compute_rtl";
+import { EntityRow, EntityConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+import { setValue } from "../../../data/input_text";
+
+@customElement("hui-input-number-entity-row")
+class HuiInputNumberEntityRow extends LitElement implements EntityRow {
+  @property() public hass?: HomeAssistant;
+  @property() private _config?: EntityConfig;
+
+  public setConfig(config: EntityConfig): void {
+    if (!config) {
+      throw new Error("Configuration error");
+    }
+    this._config = config;
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_not_found",
+            "entity",
+            this._config.entity
+          )}</hui-warning
+        >
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row
+        .hass="${this.hass}"
+        .config="${this._config}"
+        id="input_number_card"
+      >
+        <div>
+          ${stateObj.attributes.mode === "slider"
+            ? html`
+                <div class="flex">
+                  <ha-slider
+                    .dir="${this._computeRTLDirection}"
+                    .step="${Number(stateObj.attributes.step)}"
+                    .min="${Number(stateObj.attributes.min)}"
+                    .max="${Number(stateObj.attributes.max)}"
+                    .value="${Number(stateObj.state)}"
+                    pin
+                    @change="${this._selectedValueChanged}"
+                    ignore-bar-touch
+                    id="input"
+                  ></ha-slider>
+                  <span class="state">
```

Comment:
We should still hide it, but only calculate it once if it should be hidden inside `firstUpdated`.

Reply:
So I'm grabbing `this.clientWidth` in `firstUpdated` now, but it is always 0. Do I need to await something? The DOM exists at this point, so not sure why is returning 0.

Reply:
Use `connectedCallback` instead of `firstUpdated`. Then you know for sure we're attached to the DOM.

Reply:
thanks! should be all set now 🎉